src/Data/PQueue/Prio/Max.hs

```diff
@@ -298,9 +305,24 @@ foldlWithKey f z0 (MaxPQ q) = Q.foldlWithKey (\z -> f z . unDown) z0 q
 -- (@'traverseWithKey' f q == 'fromDescList' <$> 'traverse' ('uncurry' f) ('toDescList' q)@)
 --
 -- If you do not care about the /order/ of the traversal, consider using 'traverseWithKeyU'.
+--
+-- If you are working in a strict monad, consider using 'mapMWithKey'.
 traverseWithKey :: (Ord k, Applicative f) => (k -> a -> f b) -> MaxPQueue k a -> f (MaxPQueue k b)
 traverseWithKey f (MaxPQ q) = MaxPQ <$> Q.traverseWithKey (f . unDown) q
 
+-- | A version of 'traverseWithKey' designed to be efficient for strict monads
+-- such as 'IO' and strict @State@.
```

Comment:
What exactly qualifies a strict monad? That should be documented.

Reply:
A strict monad is one for which `_|_ >>= pure () = _|_`. I'm not sure that *precisely* characterizes where it makes sense to do this, TBH, but I think it's a decent heuristic. Have you looked at the code to understand what it's about?

Reply:
That link isn't working for me. What does the comment say?

Reply:
It's the one about `insertMin'`, you already replied to it.